### ISSUE TITLE
train 2class dataset from strach has nothing result

### ISSUE THREAD

**Issue by zhangyunming** [hellochick/ICNet-tensorflow#86]
hi,hellochick:
    i use the code you release severel days ago , i use my data in two class(background,foreground) to train from strach
1. the input i set 512*512
2, class_num=2,(8bit label image are 0 &1),ignore_label  set 0，later 100, 255 and so on
3, the loss is about 0.5

i test on the code in your demo.py(changed by your demo.) ,but it has nothing result.

whats wrong

**Comment by zhangyunming** [comment]
Almost all the test images are foreground..

**Comment by zhangyunming** [comment]
![icnet_tf](https://user-images.githubusercontent.com/12590457/47494844-39657300-d885-11e8-9c15-9c8a9990e269.jpg)

**Comment by 0oshowero0** [comment]
I have met the same issue. I trained my own data from scratch and got nothing output. My data has three classes, that is 0, 1 and 2, where 0 stands for background. I set n_classes to 3 & ignore 255 and n_classes to 2 & ignore 0, both got nothing. The output image is all 1.

**Comment by 0oshowero0** [comment]
Well, the code can run but it learnt nothing. When I start training, it has outputs other than 1. But later on the output is all 1.

**Comment by zhangyunming** [comment]
Yes，when i do everything same with issue50，its still the problem

**Comment by hellochick** [comment]
Do you guys load the pre-trained model, that is, use transfer learning? I think you should do that, or the network might learned nothing.

**Comment by hellochick** [comment]
Do you guys load the pre-trained model? I think you should do that, or the network might learned nothing.  If you trained from scratch, you should teach the model to classify object first (trained on ImageNet or something else).
